src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexingService.cs

```diff
@@ -297,7 +297,7 @@ await _contentDefinitionManager.AlterPartDefinitionAsync(partDefinition.Name, pa
 
                         if (included is not null && (bool)included)
                         {
-                            partDefinition.Settings.Add(nameof(ElasticContentIndexSettings), JNode.FromObject(existingPartSettings.ToObject<ElasticContentIndexSettings>()));
+                            partDefinition.Settings[nameof(ElasticContentIndexSettings)] = JNode.FromObject(existingPartSettings.ToObject<ElasticContentIndexSettings>());
```

Comment:
Would it not be better to do a check and add the field on the JsonObject like this?

if (!partDefinition.Settings.ContainsKey(nameof(ElasticContentIndexSettings)))
{
    partDefinition.Settings.Add(nameof(ElasticContentIndexSettings), JNode.FromObject(existingPartSettings.ToObject<ElasticContentIndexSettings>()));
}
else
{
    partDefinition.Settings[nameof(ElasticContentIndexSettings)] = JNode.FromObject(existingPartSettings.ToObject<ElasticContentIndexSettings>());
}

What if the setting doesn't exist at all because it has never been sync at all?

Reply:
> Would it not be better to do a check and add the field on the JsonObject like this?
> Because ; what if the setting doesn't exist at all because it has never been sync at all?

I had that pattern in my first commit of PR, @MikeAlhayek requested I change it to current version. 
I did test a clean database and it worked fine (As well as partial settings), you are setting the key regardless if it exists or not.

Reply:
@Skrypt using the code you mentions is a bit slower since it's doing the key exists first and could potentially cause concurrency issue. The latter, is slight faster and will not cause concurrency since it'll always set the value by the key.

Reply:
Ok, then that's something specific to System.Text.JSON JsonObject. That's a strange pattern still. Affecting a non existing value will create it ... That's weird.

Reply:
You can still use the .Add when using contains, but if you are using a singleton document, you could encounter issues.